src/App.js

```diff
@@ -21,33 +22,49 @@ function App() {
   };
 
   function addFavorite(id) {
-    setFavoriteIDs([...favoriteIDs, id])
+    setFavoriteIDs([...favoriteIDs, id]);
   }
 
   function removeFavorite(id) {
-    const updatedFavoriteIDs = favoriteIDs.filter((favID) => favID !== id)
-    setFavoriteIDs(updatedFavoriteIDs)
+    const updatedFavoriteIDs = favoriteIDs.filter(favID => favID !== id);
+    setFavoriteIDs(updatedFavoriteIDs);
   }
 
+  useEffect(() => {
+    localStorage.setItem('favoriteIDs', JSON.stringify(favoriteIDs));
+  }, [favoriteIDs]);
+
   useEffect(() => {
     loadCharacters();
   }, []);
 
-  
   return (
     <main>
       <Header />
       <Routes>
         <Route path="/" element={<Home characters={characters} />} />
         <Route
           path="/Details/:id"
-          element={<Details characters={characters} favoriteIDs={favoriteIDs} removeFavorite={removeFavorite}
-          addFavorite={addFavorite}  />}
+          element={
+            <Details
```

Comment:
Your Details component never uses the addFavorite and removeFavorite prop, but instead implements them again.